### ISSUE TITLE
Loss Function

### ISSUE THREAD

**Issue by EvilGhostY** [TangYJ1229/BT-SCD#1]
I have read your published paper and am very interested in the loss function discussed in it. However, I have one question and would appreciate your clarification.

In the ​​Logit_Interaction_Loss​​, the following operation is performed:
max_A, _ = torch.max(out_A, dim=1)  
max_B, _ = torch.max(out_B, dim=1)  
SCD_logit = 0.5 * (max_A + max_B)  
1. In this step, when computing max_A and max_B, are the raw logits (before softmax) used directly, rather than the probability distribution after applying softmax?
2. In semantic segmentation, the prediction output typically has a shape of ​​B×K×H×W​​ (batch × classes × height × width). Directly taking the max logits (via torch.max) discards class-specific information—meaning if both temporal phases (e.g., out_A and out_B) predict high probabilities for the ​​same class​​ at a given pixel, the resulting SCD_logit would still be large, even though this pixel is ​​not a true change region​​. Wouldn’t this make the MSE loss between SCD_logit and CD_logit.detach() somewhat counterintuitive? The loss might penalize consistent predictions (no change) simply because their raw logits are high, potentially conflicting with the actual change detection objective.
3.Regarding the dimensionality alignment, I have observed that SCD_logit has dimensions of B×H×W, whereas CD_logit.detach() exhibits dimensions of B×2×H×W. Does this imply that additional dimensionality adaptation (e.g., channel selection or pooling operations on CD_logit) is required prior to loss computation?
I would be most grateful if you could spare some time to address my questions when convenient. Thank you for your kind consideration.

**Comment by TangYJ1229** [comment]
Thank you for your interest in our work. Regarding your issue:  

1. Thank you for pointing out the issue. This was indeed an oversight in our code organization. In fact, it should be the value after the softmax operation.  

2. Since change detection is a binary classification task, we consider it to be simpler compared to semantic segmentation (a multi-class classification task). Therefore, our motivation is to leverage the change detection task to constrain the semantic segmentation branch. Specifically, for the semantic segmentation task, we predefine changed and unchanged regions (e.g., line 114 in the code), and these regions are updated with each loss computation. For the case you mentioned, if a pixel is predicted as the same category in the semantic segmentation branch, the loss will be computed using the unchanged probability from the change detection branch. However, since the change detection branch predicts this pixel as a changed class, its unchanged probability will be relatively low. This constrains the sum of the maximum probabilities from the two semantic segmentation branches to decrease, thereby encouraging the semantic segmentation branch to reduce the predicted probability for this category and reassign probabilities (recalculating for the changed region).  

3.Regarding line 117 of the code, the dimension of logit_Change is indeed B×K×H×W. The dimension of logit_Change[:, 0, :, :] is B×H×W, which is the same as the dimensions of both unchange and change. Therefore, the final CD_logit should have a dimension of B×H×W.

**Comment by EvilGhostY** [comment]
Thank you for your answer!
